examples/extra_element_integer/clustering_algorithm.h

```diff
@@ -120,10 +124,16 @@ const unsigned int FindCluster(libMesh::Mesh &mesh,
                         /*I have to pass the parent element here cause there is a big chance when doing it recursively
                          * the neighbor element would end up setting the extra_integer to the first element
                          * I have to think about a way to prevent it.
+                         * segfault error is happening in some cases due to over recursion.
+                         * pattern         1062,1063
+                         *              62,63
+                         * maybe keep track which elements are clustered temporarily?
                          */
                         neighbor_elem->set_extra_integer(index, elem->id());
```

Comment:
The fact that all this code is identical to the `ApplyRecursiveClustering` suggests that they can be combined in some way.